processor/src/chiplets/mod.rs

```diff
@@ -356,9 +380,8 @@ impl Chiplets {
         self.memory.write(ctx, addr, self.clk, new_word);
 
         // send the memory write request to the bus
-        let memory_lookup =
-            MemoryLookup::from_ints(MEMORY_WRITE_LABEL, ctx, addr, self.clk, new_word);
-        self.bus.request_memory_operation(memory_lookup, self.clk);
+        let lookup = MemoryLookup::from_ints(MEMORY_WRITE_LABEL, ctx, addr, self.clk, new_word);
```

Comment:
unrelated to the PR, but unless the user can specify the target limb via unaligned addr, this might be supersede by `write_mem`

Example:

given we have 2 words

`x = [a,b,c,d], y = [e,f,g,h]`

If I want to update `c`, then I need to use the addr of `x + 2felt`. Having that manually could imply that we need to get also the limb of the addr as argument instead of expecting the user to fix it manually.

Otherwise, if we always update only the first limb, then maybe we don't want to restore the previous word and always use 4 limbs word to save some cycles here? that would lead to bigger memory consumption.

My suggestion would be to, in the future, add an argument to specify the limb offset, and accept either `Word` or `Felt` in `write_mem` argument

Reply:
This is something we could do at the assembly level relatively easily - but at the VM level this might require significant refactoring of how constraints work.

Right now, we actually always read and write full words (4 elements) - even when we load/store a single limb to/from the stack. It is just in case of `MLoad` and `MStore` operations, all but the first element are transparent to the user. So, from cost perspective, reading a full word vs. reading just one element from it is identical.